ui/packages/catalog-ui-search/src/main/webapp/component/query-add/query-add.view.js

```diff
@@ -44,12 +45,24 @@ module.exports = Marionette.LayoutView.extend({
     'click .editor-saveRun': 'saveRun',
   },
   initialize: function() {
+    this.listenTo(user.getQuerySettings(), 'change', this.updateCurrentQuery)
```

Comment:
✏️ you could pass the 'context' to the callback here instead of grabbing it as a side effect inside it. Meaning, `this.listenTo(user.getQuerySettings(), 'change', (querySettings) => this.updateCurrentQuery(querySettings))`